daemon/api.go

```diff
@@ -234,6 +234,9 @@ func loginUser(c *Command, r *http.Request) Response {
 // It requires the state to be locked
 func UserFromRequest(st *state.State, req *http.Request) (*auth.UserState, error) {
 	// extract macaroons data from request
+	if req == nil {
```

Comment:
I don't know how often UserFromRequest is called, but I know that a lot of tests use a nil request. Maybe they should use a nullRequest instead, or something :-)